cluster/cluster.go

```diff
@@ -163,6 +164,12 @@ type ClusterRemove interface {
 
 // Cluster is the API that a cluster provider will implement.
 type Cluster interface {
+
+	// DiscoveryStart has to be called before Start and needs to be provided with
+	// a discovery implementation that could be used to bootstrap a node in the
+	// cluster
+	DiscoveryStart(discovery.Cluster) error
```

Comment:
This requirement seems awkward. Why not roll it up as a part of Start and make start accept the "discovery" thing ?